tfx/components/base/base_executor.py

```diff
@@ -160,3 +160,116 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
          output_dict: Dict[Text, List[types.Artifact]],
          exec_properties: Dict[Text, Any]) -> None:
     pass
+
+
+class BeamExecutor(BaseExecutor):
+  """ Abstract TFX executor class for handling Beam-related logic."""
+
+  @abc.abstractmethod
+  def beam_io_signature(self, input_dict: Dict[Text, List[types.Artifact]],
+                        output_dict: Dict[Text, List[types.Artifact]],
+                        exec_properties: Dict[Text, Any]
+                        ) -> List[Dict[Text, Any]]:
+    """Provides input and output signatures for the input_dict and output_dict
+       of the underlying component.
+
+    Args:
+      input_dict: Input dict from input key to a list of Artifacts. These are
+        often outputs of another component in the pipeline and passed to the
+        component by the orchestration system.
+      output_dict: Output dict from output key to a list of Artifacts. These are
+        often consumed by a dependent component.
+      exec_properties: A dict of execution properties. These are inputs to
+        pipeline with primitive types (int, string, float) and fully
+        materialized when a pipeline is constructed. No dependency to other
+        component or later injection from orchestration systems is necessary or
+        possible on these values.
+
+    Returns:
```

Comment:
If only two dicts are returned, can this be a `Tuple[List[Dict[Text, Any]], List[Dict[Text, Any]]]` instead of a Tuple?

Additionally, should `Any` be `type` instead?

Reply:
You are correct. I changed it to `Tuple[Dict[Text, type]] Dict[Text, type]]`.